yt_dlp/extractor/piapro.py

```diff
@@ -0,0 +1,97 @@
+        if not username:
+            return False
+        login_ok = True
+        login_form_strs = {
+            '_username': username,
+            '_password': password,
+            '_remember_me': 'on',
+            'login': 'ログイン'
+        }
+        self._request_webpage('https://piapro.jp/login/', None)
+        urlh = self._request_webpage(
+            'https://piapro.jp/login/exe', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(login_form_strs))
+        if urlh is False:
+            login_ok = False
+        else:
+            parts = compat_urlparse.urlparse(urlh.geturl())
+            if parts.path != '/':
+                login_ok = False
+        if not login_ok:
+            self._downloader.report_warning(
+                'unable to log in: bad username or password')
+        return login_ok
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        category_id = self._search_regex(r'categoryId=(.+)">', webpage, 'category ID')
+        if category_id not in ('1', '2', '21', '22', '23', '24', '25'):
+            raise ExtractorError('The URL does not contain audio.', expected=True)
+
+        str_duration, str_filesize = self._search_regex(
+            r'サイズ：</span>(.+?)/\(([0-9,]+?[KMG]?B)）', webpage, 'duration and size',
+            group=(1, 2), default=(None, None))
+        str_viewcount = self._search_regex(r'閲覧数：</span>([0-9,]+)\s+', webpage, 'view count', fatal=False)
+
+        str_filesize = str_filesize.replace(',', '')
+        str_viewcount = str_viewcount.replace(',', '')
+
+        uploader_id, uploader = self._search_regex(
+            r'<a\s+class="cd_user-name"\s+href="/(.*)">([^<]+)さん<', webpage, 'uploader',
+            group=(1, 2), default=(None, None))
+        content_id = self._search_regex(r'contentId\:\'(.+)\'', webpage, 'content ID')
+        create_date = self._search_regex(r'createDate\:\'(.+)\'', webpage, 'timestamp')
+
+        player_webpage = self._download_webpage(
+            f'https://piapro.jp/html5_player_popup/?id={content_id}&cdate={create_date}',
+            video_id, note='Downloading player webpage')
+
+        return {
+            'id': video_id,
+            'title': self._html_search_regex(r'<h1\s+class="cd_works-title">(.+?)</h1>', webpage, 'title', fatal=False),
+            'description': self._html_search_regex(r'<p\s+class="cd_dtl_cap">(.+?)</p>\s*<div', webpage, 'description', fatal=False),
+            'uploader': uploader,
+            'uploader_id': uploader_id,
+            'timestamp': unified_timestamp(create_date, False),
+            'duration': parse_duration(str_duration),
+            'view_count': int_or_none(str_viewcount),
+            'thumbnail': self._html_search_meta('twitter:image', webpage),
+
+            # this isn't accurate actually
+            'filesize_approx': parse_filesize(str_filesize),
```

Comment:
```suggestion
            'filesize_approx': parse_filesize(str_filesize),
```
that is why it's called approx :D